internal/rpcapi/stacks.go

```diff
@@ -765,42 +790,42 @@ func stackChangeHooks(send func(*terraform1.StackChangeProgress) error, mainStac
 		// Upon completion of a component instance plan, we emit a planned
 		// change sumary event to the client for each resource instance.
 		ReportResourceInstancePlanned: func(ctx context.Context, span any, ric *hooks.ResourceInstanceChange) any {
-			actions, err := terraform1.ChangeTypesForPlanAction(ric.Change.Action)
-			if err != nil {
-				// TODO: what do we do?
-				return span
-			}
-
 			span.(trace.Span).AddEvent("planned resource instance", trace.WithAttributes(
 				attribute.String("component_instance", ric.Addr.Component.String()),
 				attribute.String("resource_instance", ric.Addr.Item.String()),
 			))
 
-			var moved *terraform1.StackChangeProgress_ResourceInstancePlannedChange_Moved
-			if !ric.Change.PrevRunAddr.Equal(ric.Change.Addr) {
-				moved = &terraform1.StackChangeProgress_ResourceInstancePlannedChange_Moved{
-					PrevAddr: &terraform1.ResourceInstanceInStackAddr{
-						ComponentInstanceAddr: ric.Addr.Component.String(),
-						ResourceInstanceAddr:  ric.Change.PrevRunAddr.String(),
-					},
-				}
+			ripc, err := resourceInstancePlanned(ric)
+			if err != nil {
+				return span
 			}
 
-			var imported *terraform1.StackChangeProgress_ResourceInstancePlannedChange_Imported
-			if ric.Change.Importing != nil {
-				imported = &terraform1.StackChangeProgress_ResourceInstancePlannedChange_Imported{
-					ImportId: ric.Change.Importing.ID,
-				}
+			send(&terraform1.StackChangeProgress{
+				Event: &terraform1.StackChangeProgress_ResourceInstancePlannedChange_{
+					ResourceInstancePlannedChange: ripc,
+				},
+			})
+			return span
+		},
+
+		ReportResourceInstanceDeferred: func(ctx context.Context, span any, change *hooks.DeferredResourceInstanceChange) any {
+			span.(trace.Span).AddEvent("deferred resource instance", trace.WithAttributes(
```

Comment:
Just out of curiosity, why are we tracing this but not the planned?

Reply:
The planned should be traced on line 783.